### ISSUE TITLE
New Filter: remove Global filters "exclude own/found, disabled. Archived",

### ISSUE THREAD

**Issue by eddiemuc** [cgeo/cgeo#10951]
Remove the global settings for filtering status (exclude own/found, disabled, archived) and make those a pure map viewer setting (accessible via mao quick settings)

This includes removing the corresponding settings in the Settings

Last chance to protest :-)

**Comment by moving-bits** [comment]
no protest from my side here - I always found it confusing that those are related

**Comment by fm-sys** [comment]
Same for me, go for it ;-)

**Comment by Lineflyer** [comment]
Yeah, lets try it out and see if there is any gap in different use cases...

**Comment by balu2003** [comment]
I am not sure, but it seems that the global filter settings (ecxclude own/found) als has an influence of serching (MainMenu-> Search -> "Search by Owner Name". After the Search by Owner you get a list, and this **list** is filterd by the global settings.
So if you remove the menu global settings and use the "Quick Settings" instead, then the  QuickSetting is accessible **only** from the LiveMap-Quicksetting  but has influence to the search result. 
A little confusing, I think

**Comment by Lineflyer** [comment]
> So if you remove the menu global settings and use the "Quick Settings" instead, then the QuickSetting is accessible only from the LiveMap-Quicksetting but has influence to the search result.

No, those settings should then only affect the map. That of course would be the precondition to avoid such confusion. But as fas as I understood, it has been implemented in that way.

**Comment by balu2003** [comment]
For me to understand the whole "filtering" from a user view, I want to make a "chart" with all the (user) paths how  I can map a single Cache or a list, and the influence of "Qicksettings".
But it can take a few days and serves me as an overview

Main -> xxx -> Lists ->Map (the List as you see the List)
Main -> xxx -> Lists ->Map (the List as you see the List) ->LiveMap
Main -> xxx -> Single Cache -> Map the single cache
Main -> xxx -> Single Cache -> Map the single cache -> Live Map
Main->LiveMap

xxx could be Search, magnify glass, LiveMap and more ?

seems to concerns #10946

**Comment by eddiemuc** [comment]
> > So if you remove the menu global settings and use the "Quick Settings" instead, then the QuickSetting is accessible only from the LiveMap-Quicksetting but has influence to the search result.
> 
> No, those settings should then only affect the map. That of course would be the precondition to avoid such confusion. But as fas as I understood, it has been implemented in that way.

It has not yet been implemented that way, but as part of this issue it **would/must** be implemented that way. After it is completed the settings would only affect the map, no longer any other part of c:geo (like e.g. the mentioned "Search by xxx" functions)

**Comment by eddiemuc** [comment]
> For me to understand the whole "filtering" from a user view, I want to make a "chart" with all the (user) paths how  I can map a single Cache or a list, and the influence of "Qicksettings".
> But it can take a few days and serves me as an overview
> 
> Main -> xxx -> Lists ->Map (the List as you see the List)
> Main -> xxx -> Lists ->Map (the List as you see the List) ->LiveMap
> Main -> xxx -> Single Cache -> Map the single cache
> Main -> xxx -> Single Cache -> Map the single cache -> Live Map
> Main->LiveMap
> 
> xxx could be Search, magnify glass, LiveMap and more ?
> 
> seems to concerns #10946

Hi @balu2003, map viewer and it's quick settings behavior is unaffected by this issue. It will remain the only place where the filters from quick settings will still apply (exactly the way as they apply there today). Only the usage of these Settings in **all other** parts of c:geo will be removed.

**Comment by balu2003** [comment]
@eddiemuc 
you wrote

> It has not yet been implemented that way, but as part of this issue it would/must be implemented that way. After it is completed the settings would only affect the map, no longer any other part of c:geo (like e.g. the mentioned "Search by xxx" functions)>
Does this mean, that e.g. "Hidden by -> Search by Owner" lists all the Caches from this Owner in a List, an I have to filter this List with the Filter menu (icon funnel) **after** the search?

**Comment by eddiemuc** [comment]
> @eddiemuc 
> you wrote
> 
> > It has not yet been implemented that way, but as part of this issue it would/must be implemented that way. After it is completed the settings would only affect the map, no longer any other part of c:geo (like e.g. the mentioned "Search by xxx" functions)
> 
> 
> Does this mean, that e.g. "Hidden by -> Search by Owner" lists all the Caches from this Owner in a List, and I have to filter this List with the Filter menu (icon funnel) **after** the search?

Yes, but:
* The last filter you used will be applied to the result automatically (you are then able to change it of course to refilter the result)
* You may choose to use the new "search by filter" instead if you want to pre-apply a more complex filter for your online search (also here you get the last used filter prefilled). (In the long run I want to make the different "search by xxx" just prefilled variants of the "search by filter", but I haven't implement this capability for all services/connectors yet, just GC.com and oc/okapi)

**Comment by MagpieFourtyTwo** [comment]
I would prefer to leave it as it is.

Esp. excluding already found and archived caches is a general decision. And if one has once made the decision, that he doesn't want to see them (cause they are off no interest at all anymore), they should not be displayed anymore. And at least me would be really annoyed, if I would always have to add two filters, just to to get rid of stuff, I already have seen and I'm not interested in any more.

IMHO esp. suppressing found and archived caches is a basic decision, thus *is* a global setting - and has nothing to do with temporary filters applied to just a part of the app.

**Comment by eddiemuc** [comment]
> IMHO esp. suppressing found and archived caches is a basic decision, thus is a global setting -

But you are still able to do that: simply include it in your filter in the new filter system. This is applied everywhere.

(Granted, it is not yet applied in "Search by xxx" server sided, only client-sided, but this will also be implemented when I find the time)

**Comment by MagpieFourtyTwo** [comment]
> But you are still able to do that: simply include it in your filter in the new filter system. This is applied everywhere.

Yes, that's very well understood. However, I don't see any point in being forced to add two extremely basic criteria to every single new filter I will ever create. Makes no sense at all, IMHO. Would be the same as if you would ask on every opening of the map, what the limit in number of caches may be right now for add. showing the waypoints of a cache ... or which service's caches I want to see.

IMHO it is pretty easy to distinguish between general, basic decisions and those, which may change from case to case. The first should go to settings, cause I only need to make this decision once (and will meet 99% of my needs anywayl, the latter to temporary, case by case applied filters. 

And at least in my opinion, the decision if I want to see found or archive caches, is a general one. Which thus should go, resp. stay, in general settings.

**Comment by MagpieFourtyTwo** [comment]
BTW: I of course see your point WRT software development and consistent pursuit of principles. From the software point of view, of course, these *are* filters. No discussion. And would consequently have to move to this area. But iRL c:geo is not (only) used by computer scientists, on the contrary, one can certainly safely assume that most of them are quite normal users - thus usability should play a big role, too. And - IMHO - an even bigger one than software and development principles.

But back to the archived caches: They are a thing of the past. They do not even exist any longer. So why should I be forced to always remember to mask them out over and over again?

This would be like going to news.com and having to say every single time: No, I don't want to see all the old news from the last 20(!) years. No, not this time, and no, this time neither ...

Archive is archive. And - seen from the point of usability - it's way enough to say once that you don't want to see it.

Same applies IMHO to found caches. I don't need to see them anymore. Except maybe once as TJ to look up something - and then I can still show them explicitly. Once. But usually found caches are just as past as archived ones. And I don't want to be forced to have to say every single time anew that I don't want to see them ...

**Comment by balu2003** [comment]
Solution?
The global setting should remain  an should be default setting for the quicksetting or other searches
So the user can the set the global setting once and dont'd need to think about it.
But a user like me, who wants temporary see the own/found, archived .. ... caches could overwrite this default settings in the quicksetting.
Any change in the quicksetting should **not** store this settings back into global settings.
So we  all were satisfied.

**Comment by eddiemuc** [comment]
I am not sure whether we are mixing topics here. What was said above by @MagpieFourtyTwo  and @balu2003 can easily be achieved by a global filter. Right now this global filter let's you just select a cache type. It would be very easy to replace this global filter by a "full" filter where then every user can create a filter at one place which filters everywhere (in addition to any other filter applied at the place, e.g. quick settings).

In implementation right now, this is not even necessary because the same filter is always used in all places of c:geo (minus the places where I did not find the time to implement it yet, but this will follow). So this filter **IS** a global filter and has to be maintained only once. If you set e.g. a status filter to filter out "archived" and "found", then those are filtered out for whole c:geo.

So I don't understand what would be missed if I remove the existing flags "archived/found/..." in settings.

**Comment by balu2003** [comment]
Then I probably misunderstood that. If you implement this in such a way that there is a global filter setting that can be adjusted quickly and easily (in the respective context, such as the QuickSetting in LiveMap, for example), then that's good from my point of view.

**Comment by MagpieFourtyTwo** [comment]
Same for me. I thought I would have to add these to every single filter - but if it will be (resp. remains to be) a global filter which has to be set just once and will add. not appear in new filter definitions, I'm fine with this, too.

**Comment by fm-sys** [comment]
> But iRL c:geo is not (only) used by computer scientists, on the contrary, one can certainly safely assume that most of them are quite normal users - thus usability should play a big role

That's exactly the point. _Many_ users not **not** expect that the "quick settings" in the map affect totally different places in c:geo. I know quite some users (me included ;-)) who are very confused by this behavior...

**Comment by MagpieFourtyTwo** [comment]
>Right now this global filter let's you just select a cache type

In fact it was this sentence which gave the final insight to me (and I hope, it was the right one ;) ).
If the "filter" ``exclude own/found, disabled, archived``, is set at this very central point, i. e. in the main menu, where currently c:geo can globally be "restricted" to just one cache type, things sound fine to me.

Nevertheless, I would highly appreciate. if these "filters" can still be de/activated in the Quick settings menu ... and this way everything would in fact stay as it is, apart from the simple removal of the corresponding entries in the settings ... right?

**Comment by eddiemuc** [comment]
> Nevertheless, I would highly appreciate. if these "filters" can still be de/activated in the Quick settings menu ... and this way everything would in fact stay as it is, apart from the simple removal of the corresponding entries in the settings ... right?

This is where I get diverging opinions. My approach would be to make the map quick settings only apply to map viewer from now on, so when you set e.g. "exclude archived" there this would ONLY apply to map viewer.

But if you exclude "archived" in the global filter, this would be applied everywhere (including the map viewer, and not revokable through the map quick settings). 

To answer the obvious next question: it is far from trivial to "merge" the map viewer settings into a filter. Reason is that those filters can get immensely complex (might e.g. be OR filters or negated filters instead of pure positive AND filters). I would strongly suggest NOT to follow this approach. I would have no concious idea to implement it without major disruption of current filter system (which is very open and flexible intentionally), e.g. I would have to restrict it to "AND ONLY" and give status filter a "special role" throughout c:geo.

Another note: whethere there shall still be a "global filter" in the main menu explcitely (which is then applied additionally to "context-specific" filters in other places) or whether this "global filter" is just implicit (because every time you use/change a filter, the same global filter is changed and applied by default everywhere - this is how it is implemented now) is still under discussion in some other issue.

**Comment by MagpieFourtyTwo** [comment]
> This is where I get diverging opinions. My approach would be to make the map quick settings only apply to map viewer from now on, so when you set e.g. "exclude archived" there this would ONLY apply to map viewer.
> But if you exclude "archived" in the global filter, this would be applied everywhere (including the map viewer, and not revokable through the map quick settings).

Which is in fact the reason, why IMHO the map Quick Setting should still override (or even un/set) the global filter, cause otherwise you will have to change two places, to let e. g. already found caches show up in the map just once. Moreover it would not really be plausible, if ``Found`` is set to ``[X]`` in map view, but they do not show up because of the global filter ...

At the risk of repeating myself: I think the current behavior is still the best to understand. And if it needs additional explanation, one could perhaps point out in the Quick Settings popup that this changes the global filter, too ...

**Comment by eddiemuc** [comment]
> Moreover it would not really be plausible, if Found is set to [X] in map view, but they do not show up because of the global filter ...

True but it would be even less plausible if a setting in a deep-buried-place like the map quick settings would change a global filter setting (in fact, I get the impression from feedback that this even puzzles most people today).

If consistency is the most important thing. It would be best to remove the map quick settings in total and replace then with a link to the filter. However, I got the impression that most people like this special shortcut specifically for map viewer settings which is why I would leave them.

In order to implement affectiveness from map quick settings into the new filter, one would have to devise an algorithm first to integrate these Settings correctly into a possibly very complex AND/OR/NOT Filter algorithm. I don't know such an algorithm yet which doesn't destroy some of the filters users ma create.

**Comment by eddiemuc** [comment]
My suggestion right now would be to follow @Lineflyer 's advice above and try it out.

**Comment by Lineflyer** [comment]
Yes, otherwise we might end up in an endless theoretical discussion.

Lets try it and then - in case of need - open dedicated issue for use cases which got worse or are lost and try to evaluate and tackle those.
I also needed some days until I understood, that sticking to the global filter might only be cramping to known things while new approaches might still be much smoother and more powerful.

Its IMHO not a big problem is everything is different and new as long as it is mostly self explanatory and logical.

**Comment by MagpieFourtyTwo** [comment]
Ok, then please take my statement as complaint #1.  ;)

No, to be serious: This global filter will in fact be nothing else than the same global setting we currently have - with the only difference, that you can not change it on the fly, which was one of the major advantages of the Quick(!) Settings popup. Instead, it will just revert the current logic into the complete opposite, cause now, once set, it's carved in stone, And if it really overrides the choices in the map Quick Settings, it even renders all the ``Show caches`` setting in the Quick Settings popup completely useless. These will henceforth only pretend to do something - and these fake settings with no effect at all will just confuse ... and I take it for granted: Nobody will understand this hierarchy. Typical support question will be: "I activated ``Show found`` but the do not show up ...". Nobody will ever get, why one has to change *two* settings, to get found caches displayed.

Sorry, but either I still did not get your intention right, or this attempt makes no sense at all.

As for me, the Quick settings should change the global filter, too - this is, why they are *Quick* settings.

**Comment by Lineflyer** [comment]
I did not follow the complete thread. but regarding the quick settings.

The setting to e.g. hid found caches was in the past far away from being used everywhere. For example on your stored lists you always see the whole list without hiding found, disabled, archived, etc. there despite the map / global setting.
In fact we implemented various workarounds in the past to avoid users confusion is such settings would really be global without compromise.
Best example for me is, that while you might have selected to hide "owned caches" you still want to see them if you search for your owned caches. It would be confusing to see "no results".
Furthermore we often get complaints from users who say: "I searched for all caches of owner xyz, but there are some missing." The missing ones have been those already found be the user....and hidden due to the filter he believed was only for the map.

So the step of having a dedicated map view filter (quick setting) sounds like the correct one.
Again: If it turns out it has disadvantages or problematic use cases at other places, we can clarify this in the next step.

**Comment by MagpieFourtyTwo** [comment]
> So the step of having a dedicated map view filter (quick setting) sounds like the correct one.

This would be perfectly fine - but *not* if this quick setting gets mixed up with the so called global filter.

One of the main rules of GUI programming is: The user is always right. And if he explicitly demands to do s/th - then just do it!
Hence, if the user - while being in the map - says "I want to see found caches", then found caches should be shown to him.
And it would be at least confusing, if not even worse, to sarcastically grin at him and say super-teachy "Hey, but in global filter you said "no found caches", so live with it!".

And what if he has set this global filter three months ago? He will never ever guess that exactly this global filter just underhandedly destroys his aspirations. Perhaps he does not even know, where to look for it ...

But, hey, *you* are the one who has to explain to the users, who want to temporarily see found caches, why they ...

1. have to leave the map, then 
2. have to change their global filter (just to reset it immediately after anyway), then 
3. have to re-enter map, cause only then
4. the setting "Show found caches" will eventually have an effect
5. and, as mentioned before: Afterwards they have to re-enter the global filter setting, to disable "Show found caches" again, in order to not see all the already found or even archived stuff in e. g. "Nearest search" or other search results

Maybe I'm mistaken, but at least for me, this sounds like all but "straight forward" or "perfect usability".

BTW: In fact, this would just *intentionally* create exactly this error, which has just been corrected with #1100 ...

**Comment by fm-sys** [comment]
> Maybe I'm mistaken, but at least for me, this sounds like all but "straight forward" or "perfect usability".

The current behavior definitely isn't "straight forward" as well (at least for me). 

1. Search for caches of a specific owner
2. Look through the results 
2. Notice "oh, I forgot to disable the 'hide found' filter"
3. Leave the search result list
4. Leave the search screen
5. Enter the live-map
6. Open the quick settings
7. Change the filter
8. Close the quick settings
9. Close the map 
10. Open search screen again
11. Enter the user name again
12. Finally see the results you wanted to see 

So "usability" or "straight forward" is IMHO definitely even an argument for using the quick settings only in the map.

It's currently a 4:1 situation, so I would say: Let's implement it in the way how it was suggested and see how it works in real life... we can for sure try to find workarounds to improve the handling of the "duplicated found filter problem" in the map view.

**Comment by eddiemuc** [comment]
Let's try and see, and I promise I will not leave the issue after that but react / implement the insights we gain with it from user feedback (ncluding the one's gathered here so far)

Some thoughts overnight:
* maybe the name "quick settings" can be changed to e.g. "Map settings" to reflect that changes applied are "map-only". 
* in order to change the filter one doesn't have to leave the map. In fact, if such a filter is active it is displayed as a clickable "title" bar (just like today with the global filter) where one can directly click/change it. Or one can of course use the "filter" action in the action bar. Same number of clicks as for the quick settings
* One thought I had: the quick settings also allow for filtering of waypoints (parking, ...). If such a filter is set, however, this is not reflected in cache detail view (all waypoints are still shown there). Would one expect waypoints to be filtered in the cache detail view just becausehe filtered them in map view? I think "show/hide" settings for caches and waypoints should be handled consistently with each other.
* A final thought, and I can't repeat this enough: if the "quick settings" cache filter settings should be incorporated in a "new filter", I seriously need someone to think about an algorithm to do it without destroying filter logic the user has probably placed there. As an example, assume a user has a global filter `cache type = mystery OR status is not found` and now sets the "quick settings" to `show found`?

**Comment by MagpieFourtyTwo** [comment]
@fm-sys sorry but this is simply not correct. As @eddiemuc just explained, the filter can easily be reached via the header. So in lists you need just one click and you're there: you can simply change the filter, click "Go" and see an immediate effect. 

Which is the major difference to map and it's quick settings: although you can reach the filter via the header, too, there are still the checkboxes, that do absolutely nothing, until you change the filter accordingly.

TMALSS: as long as the global filter excludes several stuff, you can click the checkboxes over and over again, without any effect at all. And checkboxes, which seemingly do nothing, are considered to be broken. Mayor problem, IMNSHO.

**Comment by MagpieFourtyTwo** [comment]
@eddiemuc I still don't see the problem with seemingly colliding filters. If we just follow the usual principle: Last user action wins. Always.

So, if he decides to see found caches, they should be shown. And it should rewrite the global filter. Therefore I would maintain one single filter string, which can be changed from different places. And a checkbox in map would have the same effect as if you change the global filter. Which is IMO the most transparent way, even (or better: especially) for non experienced users.

Just a quick thought WRT the lists: The wish for list dependent settings has already been there several times, like e. g. sorting criteria, so why not just remember even a list specific filter (if stored as such explicitly)? Which will override the global filter setting in the list context only?

**Comment by eddiemuc** [comment]
> Just a quick thought WRT the lists: The wish for list dependent settings has already been there several times, like e. g. sorting criteria, so why not just remember even a list specific filter (if stored as such explicitly)? Which will override the global filter setting in the list context only?

if you want this to be pursued, would it be ok to put it into a separate issue? The whole topic "global vs local/context-specific filters" has its own complexity :-)

**Comment by eddiemuc** [comment]
> @eddiemuc I still don't see the problem with seemingly colliding filters. If we just follow the usual principle: Last user action wins. Always.

The problem lies within the usage of Non-AND-filters. Let me try to make this clearer with my example above.

Let's assume a user has defined a global filter `cache type = mystery OR status is not found`. So he would like to see only non-found caches EXCEPT for mysteries where he would also like to see found caches (let's not speculate why a user wants to do that, but it is easily possible to define such a filter). So in a nutshell he wants to see SOME found caches but not ALL of them.

Now the user opens the map quick settings

**First question:** should the "show found" checkbox now be checked or unchecked?

Now lets assume the checkbox should be unchecked (because only SOME found caches are excluded from view). Now the user checks the checkbox, so he doesn't want to see 

**Second question;** how should the filter above not be changed to reflect this new status? Different options exist:

1. Just add it as AND option on highest level: `status is not found AND (cache type = mystery OR status is not found)`. But now the resulting filter is too complex to be displayed in filter view with a simple criteria list -> would increase view complexity (/and understandability) of the filter view significantly
2. Search for "status" criteria in filter and change it's "found" setting to "not found" -> this would in current case not change the filter at all, so found caches of type mystery would still be shown
3. Search fpr "status" criteria in filter, change it's "found" setting to "not found" and change filter type to AND -> would result in `cache type = mystery AND status is not found` and thus fundamentally change the opriginal intention of the filter.

Not imagine after that the user changes the checkbox back to "not checked" -> applying the same algorithms as above will then most likely NOT leave the filter in the state it was initially.

Things get even more complex when users use negation (e.g. `NOT (cache type = mystery OR status is not found)`).

What I need is a simple, user-understandable algorithm to integrate the rather simple "exclude found from display" into potentially arbitrary complex filters working in all cases. And I don't know any. Except if we drastically reduce the allowed complexity of filters to build in general (not more "invert", no more "OR"). Is it really worth it?

**Comment by fm-sys** [comment]
> @fm-sys sorry but this is simply not correct. As @eddiemuc just explained, the filter can easily be reached via the header. So in lists you need just one click and you're there: you can simply change the filter, click "Go" and see an immediate effect.

@MagpieFourtyTwo I don't know if we are talking about different things, but in fact, filters applied in the map quick settings can currently **not** be accessed via the header!

**Comment by eddiemuc** [comment]
While trying to implement this I made one discovery, and a question arose.

The discovery I made: even today, the "exclude" options from map settings are NOT used consistently across all connectors - they are in fact only used for gc.com and OC-connectors

The question I have is about the SUConnector (API for https://geocaching.su/). It seems this API has specific URLs for searches by owner (/api/list_owner.php), by keyword (/api/list_keyword.php), by center (/api/list_center.php) and by viewport/box (/api/list.php). But what I am looking for is an API where I can combine different search parameters into one search (e.g. keyword + exclude found + viewport). Is there such a more generic search API with this service or do we really have to "choose" one single search criteria?
Where can I find documentation for this API?

**Comment by Lineflyer** [comment]
@okainov Maybe you can help here (see above)?

**Comment by eddiemuc** [comment]
PR #11015 will perform the change for "keyword" search. See there for some more thoughts.

**Comment by okainov** [comment]
>Is there such a more generic search API with this service or do we really have to "choose" one single search criteria?

Not at the moment, but I think this can be implemented. Just to be clear - you're talking about single endpoint which can optionally accept any of parameters from current endpoints, correct?

**Comment by eddiemuc** [comment]
> >Is there such a more generic search API with this service or do we really have to "choose" one single search criteria?
> 
> Not at the moment, but I think this can be implemented. Just to be clear - you're talking about single endpoint which can optionally accept any of parameters from current endpoints, correct?

Yes. Basically as much search parameters as can easily be offered.
I browsed geocaching.su (I have again problems Login in for some reason), and I discovered that in the map viewer there are already a lot of filter options which can be applied also in combination (e.g. cache type, cache size etc). Is an API available for these filters as well?

**Comment by MagpieFourtyTwo** [comment]
@eddiemuc 
> The problem lies within the usage of Non-AND-filters. Let me try to make this clearer with my example above.

Ok, complexity can of course be driven to extremes, at least theoretically, while I would have chosen an approach which would just have simplified a too complex (global) filter (maybe accompanied by an explaining toast), thus e. g. would just have removed the contradictory criteria. Which would be a valid way to handle conflicts, IMHO, having in mind that c:geo is not a relational database which must be capable of dissolving most complex boolean expressions, but is in fact just a tool to present the data of kind of weird websites in a (way) more usable way ...  ;)

But, admittedly, such simplifications can of course be confusing for a user, too.

However, if this is not a way to go, we most regretfully must capitulate to the complexity which has been created with the new (and of course nevertheless great!) filter system - which can (of course) not be mirrored to the Quick Settings menu (although I'm still convinced this is not really needed anyway, but OTOH I don't see a way to make it absolutely bullet proof at the moment either), I would suggest a completely different approach: What if we just drop the ``Show caches`` stuff from the Quick Settings menu completely, and instead offer a direct link to the new filter dialog *PLUS* a direct link to the list of saved filters, so you would be able to change the filter with two clicks ...?

Although this way we of course get exactly what I was afraid of, i. e. if you generally don't want to see found or archived caches, the criterion ``Status`` has to be included in *every* single filter (which will BTW additionally pollute the filter definition string in the header area), but this way at least any complexity of the global filter can then be mirrored to and adjusted in the map. W/o conflicts.

And perhaps the gained room can even be used to show more of the filter string ...?

BTW: I still favour just *one* single filter over the whole app, cause one filter for "all", a different for the map and a third for whatever ... will most probably confuse more than help. Apart from an explicitly set dedicated filter for the context of a list, as proposed before, in order to e. g. still be able to see found stuff in a ``Found`` list or offline logged caches in the ``Offline Logs`` list ... although the global filter would exclude these.

Apropos: The ``Status`` filter does currently not contain ``Offline logged`` although we agreed that this is in fact a state of it's own, between ``Not Found`` and ``Found``. And, yes, I know about the ``Offline Log`` and ``IS PRESENT``, but this may not be obvious to ``Basic`` users. And if we really remove the states ``Found``, ``Archived`` and ``Disabled`` from the settings, these should be added to the basic filter view, IMHO - and the ``Offline Logged`` as well.

**Comment by fm-sys** [comment]
> What if we just drop the Show caches stuff from the Quick Settings menu completely, and instead offer a direct link to the new filter dialog PLUS a direct link to the list of saved filters, so you would be able to change the filter with two clicks ...?

Somehow, I really like the idea 👍 This would indeed reduce complexity a lot...

**Comment by eddiemuc** [comment]
> BTW: I still favour just one single filter over the whole app, cause one filter for "all", a different for the map and a third for whatever ... 

With the exception of the map quick settings (and the bit yet removed global cache type filter) this is already the case

**Comment by eddiemuc** [comment]
> Apropos: The Status filter does currently not contain Offline logged although we agreed that this is in fact a state of it's own

I am behind my "implementation schedule" unfortunately, but this one can be added if course. Also "has trackables" and some other status I forgot right now are still not there. Also I wanted to introduce an own "stored since..." filter... so much to do, not nearly enough time...

**Comment by eddiemuc** [comment]
> What if we just drop the Show caches stuff from the Quick Settings menu completely, and instead offer a direct link to the new filter dialog PLUS a direct link to the list of saved filters, so you would be able to change the filter with two clicks ...?

We can do that, but be aware that this will always change the one global filter!

**Comment by eddiemuc** [comment]
> (which will BTW additionally pollute the filter definition string in the header area)

IMHO it is an error that this is not part of the definition string today, I assume users wouldn't be so surprised that their results are filtered today if the could see it there

**Comment by eddiemuc** [comment]
> What if we just drop the Show caches stuff from the Quick Settings menu completely, and instead offer a direct link to the new filter dialog PLUS a direct link to the list of saved filters, so you would be able to change the filter with two clicks ...?

Also note that users are already able to change the filter with two taps right now: you just have to tap the filter symbol in action bar of map viewer to access it. And in this view you also find the list of saved filters, if you want to use this feature.

So if I understand the request correctly it would be enough to totally drop the cache filters in map quick settings?

**Comment by Lineflyer** [comment]
> So if I understand the request correctly it would be enough to totally drop the cache filters in map quick settings?

Please keep it like it is. Its IMHO perfect now also for non advanced users. Otherwise we force everyone to use the more complex filter system for such basic tasks.

**Comment by MagpieFourtyTwo** [comment]
Which in fact would be the solution I favored, too - but as @eddiemuc stated,
filters may get too complicated and above all contradictory ...

**Comment by MagpieFourtyTwo** [comment]
> be aware that this will always change the one global filter!

Yep, intentionally. Because - not to offend anyone - but having multiple filters in mind is probably beyond the "capacity" of most users.

**Comment by fm-sys** [comment]
> > So if I understand the request correctly it would be enough to totally drop the cache filters in map quick settings?
> 
> Please keep it like it is. Its IMHO perfect now also for non advanced users. Otherwise we force everyone to use the more complex filter system for such basic tasks.

Understood. Furthermore, users might want to hide found caches only on the map, not in lists...

**Comment by Lineflyer** [comment]
> 
> Understood. Furthermore, users might want to hide found caches only on the map, not in lists...

Exactly...and also currently found, hidden, disabled, archived are not filtered out in lists which makes sense. Its now only straight forward to limit those settings to the map and offer filtering for all the rest (lists, searches).

**Comment by fm-sys** [comment]
...and if someone wants to filter found caches everywhere, he can still use the new filter framework instead of the popup (which in fact is exactly the same number of clicks as @eddiemuc had explained)

**Comment by moving-bits** [comment]
So if the map quick settings affect the map only, and get persisted for the map only, they effectively work like an additional filter which gets applied on top of the whole filter system. - Correct?

If so, we should rename the options from "show xxx" to "hide xxx" to avoid confusion, because an additional filter layer cannot show anything that's not present in the underlying filtered results, but it can hide additional elements. And maybe we can change some wording in the "popup" to make clear, that those settings will affect only the map, e. g. "Hide on map" as headline.

**Comment by eddiemuc** [comment]
> So if the map quick settings affect the map only, and get persisted for the map only, they effectively work like an additional filter which gets applied on top of the whole filter system. - Correct?

Correct, at least this is my intention.

> If so, we should rename the options from "show xxx" to "hide xxx" to avoid confusion, because an additional filter layer cannot show anything that's not present in the underlying filtered results, but it can hide additional elements. And maybe we can change some wording in the "popup" to make clear, that those settings will affect only the map, e. g. "Hide on map" as headline.

Good idea! Have to add it to the ever-increasing ToDo-list...

**Comment by fm-sys** [comment]
> they effectively work like an additional filter which gets applied on top of the whole filter system. - Correct?

Yes, at least that would be my understanding. And it should even be applied on top of the whole filter (AND) if the user had selected "global OR" in the advanced filters UI as the quick settings are in fact a totally different filter from user perspective...

**Comment by eddiemuc** [comment]
As discussed today:
* Use "context-specific" filters as follows: 
    * Offline List (for user-lists + "All" list, NOT for History)
    * Live (for live map when accessed via main menu or when map is switched to "live" mode", and for online searches e.g. Nearby, keyword etc)
    * NO persistent filter for other use cases (e.g. list/single map viewer, transient lists)
* map quick settings are merged to currenty active filter (as defined above). If filter is too complex for this (e.g. OR/NOT) then map quick settings for cache filtering are disabled
* remove global "own/found/dis/arch/offline" filter -> merge to new filters
* "Found by" search is removed from c:geo
* Bonus: make saved filters available via long-click on filter icon or current filter

![image](https://user-images.githubusercontent.com/6909759/125208816-09f52f80-e295-11eb-96d9-d8d802bdea0f.png)

**Comment by Lineflyer** [comment]
> •	remove global "own/found/dis/arch/offline" filter -> merge to new filters

Small addition: 
Throw away existing filters of beta users if needed for migration. 

Remark to myself:
Explicitly test migration path from last release to next beta later on

**Comment by MagpieFourtyTwo** [comment]
Some more smalls:
- Terminology: "ONLINE" filter => "Live" filter
- Filter dialog header += indicator/keyword for context, e. g. ``Filter for Live data`` resp ``Filter for Offline data`` or alike
- Long click on Filter definition string (below header) => open "Saved Filter List" (analog to long click on filter icon)

**Comment by fm-sys** [comment]
> Small addition:
> Throw away existing filters of beta users if needed for migration.

Migrate the existing filter to "live" and/or "offline"? I would even say let's throw an existing filter in away in all cases for beta users, as we can't do it right... That makes things cleaner, and makes it more easy to prefill the live filter with the map quick settings.

**Comment by Lineflyer** [comment]
> Migrate the existing filter to "live" and/or "offline"?

Migrate only to "live", not to "offline"!

**Comment by moving-bits** [comment]
> Filter dialog header += indicator/keyword for context, e. g. `Filter for Live data` resp `Filter for Offline data` or alike

We should try to keep that indicator short to not fill up too much screen estate, so maybe only "live" and "offline" as prefix?

**Comment by Lineflyer** [comment]
`Live Filter` and `Offline filter` sound good IMHO

**Comment by eddiemuc** [comment]
> > Small addition:
> > Throw away existing filters of beta users if needed for migration.
> 
> Migrate the existing filter to "live" and/or "offline"? I would even say let's throw an existing filter in away in all cases for beta users, as we can't do it right... That makes things cleaner, and makes it more easy to prefill the live filter with the map quick settings. 

I plan a "best-effort" migration: if existing Filter is an AND filter I migrate into it (I need this logic anyway for "map quick settings"), otherwise I throw the existing filter away.

**Comment by eddiemuc** [comment]
Well, looks like some work to do for me during next weeks.

**Comment by eddiemuc** [comment]
PR #11252 should now be ready
